gpuallocator/device.go

```diff
@@ -7,61 +7,128 @@ import (
 	"sort"
 	"strings"
 
-	"github.com/NVIDIA/gpu-monitoring-tools/bindings/go/nvml"
+	"github.com/NVIDIA/go-gpuallocator/internal/links"
+	"github.com/NVIDIA/go-nvlib/pkg/nvlib/device"
+	"github.com/NVIDIA/go-nvlib/pkg/nvml"
 )
 
 // Device represents a GPU device as reported by NVML, including all of its
 // Point-to-Point link information.
 type Device struct {
-	*nvml.Device
+	device.Device
 	Index int
 	Links map[int][]P2PLink
+	// The previous binding implementation used to cache specific device properties.
+	// These should be considered deprecated and the functions associated with device.Device
+	// should be used instead.
+	cachedDeviceInfo
+}
+
+// The previous binding implementation used to cache specific device properties.
+// We collect these into a separate type to allow for embedding and simpler migration from them.
+type cachedDeviceInfo struct {
+	UUID string
+	PCI  struct {
+		BusID string
+	}
+	CPUAffinity *uint
+}
```

Comment:
I would have expected it to be like:
```
type Device struct {
	nvlibDevice
	Index int
	Links map[int][]P2PLink
}

type nvlibDevice struct {
	device.Device
	// The previous binding implementation used to cache specific device properties.
	// These should be considered deprecated and the functions associated with device.Device
	// should be used instead.
	UUID string
	PCI  struct {
		BusID string
	}
	CPUAffinity *uint
}
```

Reply:
OK. Will update. One thing I was just wondering was whether `Index` should also be in the `nvlibDevice` struct, but I'm going to leave it out for now.